src/tutorials/konnector/2fa.md

```diff
@@ -0,0 +1,62 @@
+You have successfully developped your connector but you realize that some accounts need two factor
+authentication (2FA) to work.
+
+Here is how you can add this to your connector
+
+## 2FA detection
+
+Every target website being different, you will have to develop it yourself. Some websites return a
+specific url, some others display a special html content.
+
+## Home Automatic Successull Login deactivation
+
+By default, the home will suppose the connector's login is a success after 8s, which is bad when
+your connector needs to wait for the user to send his code.
+
+For this, call :
+
+```javascript
+await this.deactivateAutoSuccessfulLogin()
+```
+
+where `this` is your connector instance.
+
+And when the 2FA process is finished, call :
+
+```javascript
+await this.notifySuccessfulLogin()
+```
+
+To notify the home that the login process is a success.
+
+## Request a 2FA code from the user
+```javascript
+const code = await this.waitForTwoFaCode()
+```
+
+![Home 2FA popup](./2fa_popup.png)
+
+Which will display this popup in home and return the code when ready.
+It will handle error messages on timeout and won't request a 2FA code if the connector is run in
+background and throw the `USER_ACTION_NEEDED.TWOFA_EXPIRED` error instead.
+
+You can get more information [in the documentation](https://docs.cozy.io/en/cozy-konnector-libs/api/#basekonnectorwaitfortwofacode-promise)
+
+When you code your code, send it to the target website the way it wants
```

Comment:
« When you get your code »? Maybe even « When you get the 2FA code ». I don't know if I understand what you wanted to say.